simpeg/directives/_save_geoh5.py

```diff
@@ -503,3 +505,64 @@ def get_names(
             base_name = "LP models"
 
         return channel_name, base_name
+
+
+class SavePGIModel(SaveArrayGeoH5):
+    """
+    Save the model as a property group in the geoh5 file
+    """
+
+    def __init__(
+        self,
+        h5_object,
+        pgi_reg: PGIsmallness,
+        unit_map: dict,
+        physical_properties: list[str],
+        value_map: dict[int, str] | None = None,
+        **kwargs,
+    ):
+        self.pgi_reg = pgi_reg
+        self.unit_map: dict = unit_map
+        self.value_map = value_map
+        self.physical_properties = physical_properties
+        super().__init__(h5_object, **kwargs)
+
+    def get_values(self, values: list[np.ndarray] | None):
+
+        if values is None:
+            values = self.invProb.model
+
+        modellist = self.pgi_reg.wiresmap * values
+        model = np.c_[[a * b for a, b in zip(self.pgi_reg.maplist, modellist)]].T
```

Comment:
is np.c_[something].T not the same as np.r_[something]?

Reply:
No cause np.r_[] is equivalent as an hstack, so all models get stack along one long vector. The np.c_[] creates a 2D array.

Reply:
gotcha